init.c

```diff
@@ -2190,6 +2210,18 @@ static int parse_set(struct Buffer *buf, struct Buffer *s, unsigned long data,
           char scratch[PATH_MAX];
           mutt_str_strfcpy(scratch, buf->data, sizeof(scratch));
           mutt_expand_path(scratch, sizeof(scratch));
+
+          if ((url_check_scheme(scratch) == U_UNKNOWN) /* probably a local file */
+              && (strlen(scratch) > 0)
+              && (scratch[strlen(scratch) - 1] != '|')) /* Is this a file or a command meant to be run? */
```

Comment:
hehe.  Better, but...
Now we run `strlen()` twice, which is a little inefficient.